### PR TITLE
feat(TDOPS-4789/facetedSearch): callback support for BadgeMenu

### PR DESCRIPTION
**Pull request by ybaskaran**
**What is the problem this PR is trying to solve?**
callback support for BadgeMenu

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/faceted-search/src/components/Badges/BadgeMenu/BadgeMenu.component.js [hunk 1]
```diff
@@ -1,11 +1,18 @@
-import { useMemo } from 'react';
+import { useEffect, useMemo, useState } from 'react';
+
 import { isEmpty } from 'lodash';
```
Comment (inna-i) [inline packages/faceted-search/src/components/Badges/BadgeMenu/BadgeMenu.component.js:3]:
```suggestion
import isEmpty from 'lodash/isEmpty';
```

**Review by inna-i — APPROVED** [review]

File: packages/faceted-search/src/components/Badges/BadgeMenu/BadgeMenu.component.js [hunk 2]
```diff
@@ -31,10 +38,37 @@ export const BadgeMenu = ({
 	displayType,
 	filterBarPlaceholder,
 	t,
+	callbacks,
 	...rest
 }) => {
+	const [options, setOptions] = useState(values);
 	const currentOperators = useMemo(() => operators, [operators]);
 	const currentOperator = operator || (currentOperators && currentOperators[0]);
+	const [isLoading, setIsLoading] = useState(true);
```
Comment (yyanwang) [inline packages/faceted-search/src/components/Badges/BadgeMenu/BadgeMenu.component.js:47]: maybe better to set isLoading false by default? then no need for this line: https://github.com/Talend/ui/pull/5022/files#diff-9cd7307a01bfb12fcee969fa3d4f76874a44bf7dc27b1c37d95e3157942ee263R51

File: packages/faceted-search/src/components/Badges/BadgeMenu/BadgeMenuForm.component.js [hunk 3]
```diff
@@ -119,6 +127,8 @@ const BadgeMenuForm = ({
 						type="submit"
 						label={t('APPLY', { defaultValue: 'Apply' })}
 						bsStyle="info"
+						disabled={rest.isLoading}
+						inProgress={rest.isLoading}
```
Comment (yyanwang) [inline packages/faceted-search/src/components/Badges/BadgeMenu/BadgeMenuForm.component.js:131]: IMO no need set `inProgress` to true when loading options, just disable submit button is enough. set `inProgress` to true when it's submitting.

File: packages/faceted-search/src/components/facetedSearch.propTypes.js [hunk 4]
```diff
@@ -39,6 +39,7 @@ const badgeFacetedPropTypes = PropTypes.shape({
 		isInCreation: PropTypes.bool,
 		entitiesPerBadge: PropTypes.string,
 		operators: PropTypes.arrayOf(PropTypes.string),
+		isLoading: PropTypes.bool,
```
Comment (yyanwang) [inline packages/faceted-search/src/components/facetedSearch.propTypes.js:42]: Do we need to add isLoading to metadata?

Comment (ybaskaran) [reply packages/faceted-search/src/components/facetedSearch.propTypes.js]: removed

File: packages/faceted-search/src/components/Badges/BadgeMenu/BadgeMenuForm.component.js (hunk 3 above)
Comment (ybaskaran) [reply packages/faceted-search/src/components/Badges/BadgeMenu/BadgeMenuForm.component.js]: removed

File: packages/faceted-search/src/components/Badges/BadgeMenu/BadgeMenu.component.js (hunk 2 above)
Comment (ybaskaran) [reply packages/faceted-search/src/components/Badges/BadgeMenu/BadgeMenu.component.js:47]: fixed

File: packages/faceted-search/src/components/Badges/BadgeMenu/BadgeMenu.component.test.js [hunk 5]
```diff
@@ -53,4 +55,57 @@ describe('BadgeMenu', () => {
...
+			<BadgeFacetedProvider value={badgeFacetedContextValue}>
+				<BadgeMenu {...props} />
+			</BadgeFacetedProvider>,
+		);
+
+		// Then there is a checkbox with data taken from callback
+		await waitFor(() => {
+			expect(screen.getByRole('menuitem')).toBeVisible();
+		});
+		screen.logTestingPlaygroundURL();
```
[the remark this answers is not in the document]
Comment (ybaskaran) [reply packages/faceted-search/src/components/Badges/BadgeMenu/BadgeMenu.component.test.js]: removed

**Pull request merged**